protoc-gen-openapiv2/internal/genopenapi/template.go

```diff
@@ -1145,6 +1156,78 @@ func renderServiceTags(services []*descriptor.Service, reg *descriptor.Registry)
 	return tags
 }
 
+// expandPathPatterns search the URI parts for path parameters with pattern and when the pattern contains a sub-path,
+// it expands the pattern into the URI parts and adds the new path parameters to the pathParams slice.
+//
+// Parameters:
+//   - pathParts:	the URI parts parsed from the path template with `templateToParts` function
+//   - pathParams: the path parameters of the service binding, this slice will be mutated when the path pattern contains
+//     a sub-path with wildcard.
+//
+// Returns:
+//
+//	The modified pathParts. Also mutates the pathParams slice.
+func expandPathPatterns(pathParts []string, pathParams *[]descriptor.Parameter) []string {
```

Comment:
Since this mutates an input, what do you think about returning a parameter of the same type instead? It's a bit gross IMO to mutate the input slice and returning it will make it more explicit.
```suggestion
//	The modified pathParts and pathParams slice.
func expandPathPatterns(pathParts []string, pathParams []descriptor.Parameter) ([]string, []descriptor.Parameter) {
```